### PR TITLE
Support capitalized day and month names

### PR DESCRIPTION
**Pull request by mrossinek**
As pointed out in https://github.com/GothenburgBitFactory/taskwarrior/issues/2160 it used to be possible to use capitalized day and month names in Taskwarrior. Since switching to the `Datetime` format in https://github.com/GothenburgBitFactory/taskwarrior/commit/434e95f560b025f07d2fa5fb9a3733fa1ddb4d42, this is no longer possible.
This commit tries to fix that by ensuring the `Pig::skipPartial` method always compares lower-case characters (since this is what `reference` always contains).

Unittests are added to assert this behavior.

### CODE REVIEW

**Comment by mrossinek** [comment]
@tbabej I saw some activity over on Taskwarrior by you. Would you mind also taking a look here?

File: src/Pig.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ bool Pig::skipPartial (const std::string& reference, std::string& result)
   auto pos = 0;
   while (reference[pos] &&
          (*_text)[_cursor + pos] &&
-         reference[pos] == (*_text)[_cursor + pos])
+         reference[pos] == tolower((*_text)[_cursor + pos]))
```
Comment (tbabej) [inline src/Pig.cpp:112]: I don't think modifying the behaviour of `skipPartial` is the way to go. While this is currently only used for detecting the named dates in `Datetime.cpp`, the name of the method does not suggest that the comparison is based on case-insensitive search. Its counterpart - the `skipLiteral` method - is not case insensitive either.

What I'd suggest is either implementing a new method for this (`skipPartialInsensitive`?), or adding a boolean flag here.

**Review by tbabej — CHANGES_REQUESTED** [review]
Additionally, we should think about how case insensitivity fits into the rest of the ecosystem. If we allow arbitrary capitalization of the datatime names, are there other attributes with built-in values for attributes where capitalization matters?

File: src/Pig.cpp (hunk 1 above)
Comment (mrossinek) [reply src/Pig.cpp:112]: Yes that makes sense! I will take a closer look either later today or later this week :+1:

**Comment by mrossinek** [comment]
> Additionally, we should think about how case insensitivity fits into the rest of the ecosystem. If we allow arbitrary capitalization of the datatime names, are there other attributes with built-in values for attributes where capitalization matters?

Currently I can only think of weekdays and months where this might matter. :thinking: I am not sure how we expect to handle user-defined attributes like project names, tags, etc.. Although I am also not sure whether this gets handled by `libshared`.

**Comment by mrossinek** [comment]
I added the optional `ignore_case` argument to `skipPartial`. Do you want a similar option on `skipLiteral`?

I also grepped for `"[A-Z]` on the `test` directory and could not find any other strings which we might be interested in also supporting lower-case versions of. However, this obviously does not cover all possible cases.

**Review by tbabej — APPROVED** [review]

**Comment by tbabej** [comment]
> I also grepped for "[A-Z] on the test directory and could not find any other strings which we might be interested in also supporting lower-case versions of.

Tests are not exhaustive, unfortunately, but if you grep for `skipLiteral`, you'll find a bunch of named Datetime constants:
```
...
src/Datetime.cpp:  if (pig.skipLiteral ("now"))
src/Datetime.cpp:  if (pig.skipLiteral ("sopd"))
src/Datetime.cpp:  if (pig.skipLiteral ("sod"))
src/Datetime.cpp:  if (pig.skipLiteral ("sond"))
src/Datetime.cpp:  if (pig.skipLiteral ("eopd"))
src/Datetime.cpp:  if (pig.skipLiteral ("eod"))
src/Datetime.cpp:  if (pig.skipLiteral ("eond"))
src/Datetime.cpp:  if (pig.skipLiteral ("sopw"))
...
```

**Pull request merged**